components/core/src/ffi/ir_stream/decoding_methods.cpp

```diff
@@ -292,8 +295,6 @@ namespace ffi::ir_stream {
         }
 
         // Handle variables
-        vector<encoded_variable_t> encoded_vars;
-        vector<string> dict_vars;
         encoded_variable_t encoded_variable;
         string var_str;
```

Comment:
Can we move encoded_variable and var_str into the loop body to tighten the scope? I know we might reuse var_str's low-level memory space, but I doubt it will have an impact on the performance

Reply:
moved encoded_variable  into the loop body but still keep var_str out side of the loop.